src/JsonSchema/Constraints/ConstraintInterface.php

```diff
@@ -59,5 +59,10 @@ public function isValid();
      * @param mixed            $i
      * @throws \JsonSchema\Exception\ExceptionInterface
      */
+    public function _check(&$value, $schema = null, JsonPointer $path = null, $i = null);
```

Comment:
I guess another option would be moving the contents of `Validator::check()` into `validate()`, and leaving `_check()` as an empty method (it can't be removed without also removing it from the interface, which is an important sanity check for those writing constraints). I don't much mind either way - would having `Validator::_check()` be a noop be preferable for you?